libraries/AP_Mission/AP_Mission.cpp

```diff
@@ -2374,14 +2387,62 @@ bool AP_Mission::jump_to_landing_sequence(const Location &current_loc)
         }
 
         GCS_SEND_TEXT(MAV_SEVERITY_INFO, "Landing sequence start");
-        _flags.in_landing_sequence = true;
         return true;
     }
 
     GCS_SEND_TEXT(MAV_SEVERITY_WARNING, "Unable to start landing sequence");
     return false;
 }
 
+/*
+   find the closest point on the mission after a DO_RETURN_PATH_START and before DO_LAND_START or landing
+ */
+bool AP_Mission::jump_to_closest_mission_leg(const Location &current_loc)
+{
+    if (_flags.state == MISSION_RUNNING) {
+        // if mission is already running don't switch away from a active landing or return path
+        if (_flags.in_landing_sequence) {
+            GCS_SEND_TEXT(MAV_SEVERITY_INFO, "Landing sequence active");
```

Comment:
Ideally we should have the subsystem prefixed on the text messages I think.  I suspect we have other examples in the mission library that don't use a prefix.. we probably should though.  Not a blocker though.